File: stress/t/arr_runmax.f
      PROGRAM ARRRUNMAX
      INTEGER*8 I,N,M,MI,T,CHK
      INTEGER*8 A(0:19),R(0:19)
      N=20
      I=0
   10 IF (I.GE.N) GOTO 20
      A(I)=MOD(I*13+5,29_8)-10
      I=I+1
      GOTO 10
   20 CONTINUE
      M=A(0)
      I=0
   30 IF (I.GE.N) GOTO 40
      T=A(I)
      IF (T.GT.M) M=T
      R(I)=M
      I=I+1
      GOTO 30
   40 CONTINUE
      M=A(0)
      MI=0
      I=1
   50 IF (I.GE.N) GOTO 60
      T=A(I)
      IF (T.GT.M) THEN
        M=T
        MI=I
      ENDIF
      I=I+1
      GOTO 50
   60 CONTINUE
      CHK=R(19)
      WRITE(*,'(A,I0)') 'runmax_last=',CHK
      CHK=R(10)
      WRITE(*,'(A,I0)') 'runmax10=',CHK
      WRITE(*,'(A,I0)') 'maxval=',M
      WRITE(*,'(A,I0)') 'maxidx=',MI
      END

File: stress/t/arr_runmax.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t i,n,m,mi,t,chk;
  int64_t A[20],R[20];
  n=20;
  for(i=0;i<n;i++){ A[i]=(i*13+5)%29-10; }
  m=A[0];
  for(i=0;i<n;i++){ t=A[i]; if(t>m){ m=t; } R[i]=m; }
  m=A[0]; mi=0;
  for(i=1;i<n;i++){ t=A[i]; if(t>m){ m=t; mi=i; } }
  chk=R[19]; printf("runmax_last="); printf("%ld",(long)chk); printf("\n");
  chk=R[10]; printf("runmax10="); printf("%ld",(long)chk); printf("\n");
  printf("maxval="); printf("%ld",(long)m); printf("\n");
  printf("maxidx="); printf("%ld",(long)mi); printf("\n");
  return 0;
}
